core/src/components/popover/popover.tsx

```diff
@@ -434,17 +435,7 @@ export class Popover implements ComponentInterface, PopoverInterface {
       return;
     }
 
-    /**
-     * When using an inline popover
-     * and dismissing a popover it is possible to
-     * quickly present the popover while it is
-     * dismissing. We need to await any current
-     * transition to allow the dismiss to finish
-     * before presenting again.
-     */
-    if (this.currentTransition !== undefined) {
-      await this.currentTransition;
-    }
+    const unlock = await this.lockController.lock();
```

Comment:
For modal and popover we'll want to acquire the lock before we do the `this.presented` check otherwise the modal will just return early if you try to present, dismiss, then present quickly.